addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -0,0 +1,189 @@
+/** @odoo-module **/
+
+import {generateHTMLId} from 'web_editor.utils';
+import options from 'web_editor.snippets.options';
+
+options.registry.SocialMedia = options.Class.extend({
+    /**
+     * @override
+     */
+    async willStart() {
+        this.$target[0].querySelectorAll('a').forEach(el => el.setAttribute('id', generateHTMLId(30)));
+        await this._super(...arguments);
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+        // Fetch URLs for db links.
+        [this.dbSocialValues] = await this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [websiteId, ['social_facebook', 'social_twitter', 'social_youtube',
+                'social_instagram', 'social_linkedin', 'social_github']],
+        });
+        delete this.dbSocialValues.id;
+    },
+    /**
+     * @override
+     */
+    async cleanForSave() {
+        this.$target[0].querySelectorAll('a').forEach(el => el.removeAttribute('id'));
+        // Update the DB links.
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+        this._rpc({
+            model: 'website',
+            method: 'write',
+            args: [[websiteId], this.dbSocialValues],
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Applies the we-list on the target and rebuilds the social links.
+     *
+     * @see this.selectClass for parameters
+     */
+    async renderListItems(previewMode, widgetValue, params) {
+        const entries = JSON.parse(widgetValue);
+        // Handle element deletation.
+        const entriesIds = entries.map(entry => entry.id);
+        const anchorsEls = this.$target[0].querySelectorAll('a');
```

Comment:
The 'a' selector is a problem, the user can put the title of the block as a link and break the whole we-list. I think the easiest way is to add a class like s_social_media_link and to select only the links that have this class.

Reply:
I am not a big fan of adding classes everywhere, even if it is more robust indeed.
I'd personnally do something like `:scope > a` as selector instead of `a` then, as there is no way to add a link as an immediate child, so we will always get what we expect: the links of the social media, right?